docs/rfc/active/data-permissions/README.md

```diff
@@ -0,0 +1,138 @@
+- Start Date: 2022-04-19
+- RFC PR: #4694
+- Discussion Issue: 
+- Implementation PR(s): 
+
+# Modeling Data Permissions
+
+## Summary
+
+Many datahub sources provide access control mechanisms to manage user access to assets.
+Datahub should provide functionality for viewing these permissions.
+
+## Motivation
+
+Within many sql systems, the data control language provides commands like GRANT, DENY and REVOKE to manage control to datasets.
+
+Being able to view grants is the first step toward supporting larger integrations with dataset permissions. [See Here](https://feature-requests.datahubproject.io/b/feedback/p/access-control-on-datasets-bi-assets)
+
+Of the datahub integration sources listed, the following have some form of data access control management:
+
+- Athena (via IAM)
+- BigQuery
+- Clickhouse
+- Druid
+- Elastic Search
+- Glue
+- Hive (via Apache Ranger)
+- Kafka
+- Looker
+- MariaDB
+- Metabase
+- MongoDB
+- MicrosoftSQLServer
+- MySQL
+- Oracle
+- PostgreSQL
+- Redash
+- Redshift
+- S3 (via IAM)
+- Sagemaker
+- Snowflake
+- Superset
+- Tableau
+- Trino
+
+## Requirements
+
+The permissions aspect must be:
+
+- Appliable to datasets, containers, dashboards
+- Support ingesting and viewing structured access control (GRANT {permission} ON {object} TO {user or role}
+- Support ingesting and viewing unstructured access control (IAM Policy Documents, Custom Policies from other datasets)
+-
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding
+> a new API or set of models that others can build on in later? Please list these concerns here as well.
+
+## Non-Requirements
+
+This RFC specifically calls for functionality to view the permissions. Actions taken on the permissions may be great areas for future enhancement but is not a focus here.
+
+## Detailed design
+
+There are no new Entities associated with this design.
```

Comment:
Agree with Shirshanka's comment; also, we may need to evolve the kinds of "structured" primitives/clauses that we model over time.

So we could create an entity, say, DataPermission.

Simplistically, it could contain below optional fields:
- resource captured as arbitrary urn
- principal i.e. user/role
- tag (glossary term) key-value (linkage to glossary model TBD)
- permission string
- native (equivalent to "unstructured"; we conventionally use the term "native" in such cases)

Depending on the nature of the permission, only appropriate fields would be filled. 
A more rigorous, extensible, implementation can separately categorize ACL, RBAC, tag-based and native records using a discriminator. (We could later add, say, simple masking policies as an additional structured record.)

This will allow us to traverse the metadata graph to find, for example, a column/table's policy tag -> roles -> users by traversing the corresponding DataPermission, CorpRole, CorpUser entities.

Finally, it may be worth looking at Datahub's existing DataHubPolicyInfo model; it may be possible to merge the concepts of this new proposed DataPermission with the existing model with some minimal discriminator/genesis fields to indicate whether an entry is an externally sourced data-permission or a datahub-native policy. I'm not sure this is a good idea as these are conceptually somewhat different; but their significant structural similarity might encourage reuse in model impl.